### PR TITLE
Lesson01 & Lesson 02 commits

### PR DESCRIPTION
**Pull request by russlarge256**

### CODE REVIEW

File: students/Russell_Large/template_student/lesson01/Submittal/Testing/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# inventory_management
+# Russ Large, created script (4/16/2019)
+
+from unittest import TestCase
+from unittest.mock import patch
+import unittest.mock
+import sys
+
+# append system path, so main can be imported as 'import main'
+sys.path.append(r'C:\Python220\Lesson01\assignment\inventory_management')
```
Comment (majstorpoki) [inline students/Russell_Large/template_student/lesson01/Submittal/Testing/test_integration.py:12]: Good!

File: students/Russell_Large/template_student/lesson02/assignment/src/charges_calc.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                        .format(parser.parse_args()))
+
+def load_rentals_file(filename):
+    '''This function will take all calculated data and add it
+       to the specified args file'''
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+        except ValueError:
+            exit(0)
```
Comment (majstorpoki) [inline students/Russell_Large/template_student/lesson02/assignment/src/charges_calc.py:64]: Any logging?

File: students/Russell_Large/template_student/lesson02/assignment/src/charges_calc.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            value['total_days'] = (rental_end - rental_start).days
+            value['total_price'] = value['total_days'] * value['price_per_day']
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+            value['unit_cost'] = value['total_price'] / value['units_rented']
+
+            logging.debug("{} has been calculated. ".format(value['product_code']))
+
+
+        except ValueError:
+            logging.error("incorrect value for rental_start: {} or rental_end: {}"
```
Comment (majstorpoki) [inline students/Russell_Large/template_student/lesson02/assignment/src/charges_calc.py:82]: Is this the ONLY reason ValueError will be triggered? How to be more specific?

File: students/Russell_Large/template_student/lesson03/assignment/src/basic_operations.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                      'cust_phone': query.cust_phone
+                      }
+
+        logger.info(f'The following cutomer information has been selected:'
+                    f'{query_dict}')
+
+        return query_dict
+
+    except Exception as e:
+        logger.info('search customer unsuccessful.')
```
Comment (majstorpoki) [inline students/Russell_Large/template_student/lesson03/assignment/src/basic_operations.py:73]: Search unsuccessful or a record cannot be found?!

File: students/Russell_Large/template_student/lesson03/assignment/src/basic_operations.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    exception if the customer does not exist
+    :param customer_id:
+    :param credit_limit:
+    :return:
+    '''
+
+    try:
+        with database.transaction():
+            q = (Customer
+                 .update(cust_credit_limit = credit_limit)
```
Comment (majstorpoki) [inline students/Russell_Large/template_student/lesson03/assignment/src/basic_operations.py:109]: I like how you did this, by directly writing an update query

File: students/Russell_Large/template_student/lesson03/assignment/src/basic_operations.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+    ##Update##
+    # changed query to search for 'Active', not 'active'
+    # csv has actives as 'Active'
+    
+    try:
+        with database.transaction():
+            query = (Customer.select()
+                     .where(Customer.cust_status == 'Active'))\
+                     .count()
```
Comment (majstorpoki) [inline students/Russell_Large/template_student/lesson03/assignment/src/basic_operations.py:143]: I like this solution

**Review by majstorpoki — APPROVED** [review]
I specifically like your Assignment 3 solution.

**Review by ObjectiveJoe — APPROVED** [review]

**Pull request merged**